lib/_tls_wrap.js

```diff
@@ -240,6 +245,46 @@ function onnewsession(key, session) {
     done();
 }
 
+function onPskServerCallback(identity, maxPskLen) {
+  const owner = this[owner_symbol];
+  const psk = owner[kPskCallback](identity);
+  if (psk == null) {
+    return undefined;
+  } else {
+    if (typeof psk !== 'string') {
+      throw new ERR_INVALID_ARG_TYPE('psk', ['string']);
+    }
+    if (psk.length / 2 > maxPskLen) {
+      throw new ERR_ASSERTION(`Pre-shared key exceeds ${maxPskLen} bytes`);
+    }
+    return psk;
+  }
+}
+
+function onPskClientCallback(hint, maxPskLen, maxIdentityLen) {
+  const owner = this[owner_symbol];
+  const ret = owner[kPskCallback](hint);
+  if (typeof ret !== 'object' || ret == null) {
+    return undefined;
+  } else {
+    const result = { psk: ret.psk, identity: ret.identity };
+
+    if (typeof result.psk !== 'string') {
+      throw new ERR_INVALID_ARG_TYPE('psk', ['string']);
+    }
+    if (result.psk.length / 2 > maxPskLen) {
+      throw new ERR_ASSERTION(`Pre-shared key exceeds ${maxPskLen} bytes`);
+    }
+
+    if (typeof ret.identity !== 'string') {
+      throw new ERR_INVALID_ARG_TYPE('identity', 'string');
+    }
+    if (ret.identity.length > maxIdentityLen) {
+      throw new ERR_ASSERTION(`PSK identity exceeds ${maxIdentityLen} bytes`);
```

Comment:
Isn't `string.length` in unicode characters, not bytes, so when encoded as bytes it could be longer than maxIdentityLen bytes?